src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs

```diff
@@ -66,6 +68,15 @@ protected static void InitCache(IDbConfig dbConfig)
         }
     }
 
+    private static readonly Dictionary<string, string> MissingNativeLibraryNameMapping = new() { { "libdl", "libdl.so.2" } };
+
+    private static IntPtr OnResolvingUnmanagedDll(Assembly _, string nativeLibraryName) =>
+        MissingNativeLibraryNameMapping.ContainsKey(nativeLibraryName)
```

Comment:
btw its just one, why not hardcode instead of dictionary?

Reply:
I also think we can hardcode this for now although the dictionary gives us the flexibility to add other libs as well. But for now, looks like we don't need that kind of flexibility.